test_util/test_upgrade_vpc.py

```diff
@@ -159,125 +132,277 @@ def wait_for_dns(hostname):
+                ),
+            )
+            vpc.wait_for_complete()
+
+        cluster = test_util.cluster.Cluster.from_vpc(
+            vpc,
+            ssh_info,
+            ssh_key=self.ssh_key,
+            num_masters=self.num_masters,
+            num_agents=self.num_agents,
+            num_public_agents=self.num_public_agents,
+        )
+
+        with logger.scope("install dcos"):
+            # Use the CLI installer to set exhibitor_storage_backend = zookeeper.
+            test_util.cluster.install_dcos(cluster, self.stable_installer_url, api=False,
+                                           add_config_path=self.config_yaml_override_install)
+
+            master_list = [h.private_ip for h in cluster.masters]
+
+            dcos_api_install = self.dcos_api_session_factory_install.apply(
+                'http://{ip}'.format(ip=cluster.masters[0].public_ip),
+                master_list,
+                master_list,
+                [h.private_ip for h in cluster.agents],
+                [h.private_ip for h in cluster.public_agents],
+                self.default_os_user)             # default_os_user
+
+            dcos_api_install.wait_for_dcos()
+
+        healthcheck_app = create_marathon_healthcheck_app(healthcheck_app_id)
+        dns_app = create_marathon_dns_app(dns_app_id, healthcheck_app_id)
+
+        self.setup_cluster_workload(dcos_api_install, healthcheck_app, dns_app)
+
+        with logger.scope("upgrade cluster"):
+            test_util.cluster.upgrade_dcos(cluster, self.installer_url,
+                                           add_config_path=self.config_yaml_override_upgrade)
+            with cluster.ssher.tunnel(cluster.bootstrap_host) as bootstrap_host_tunnel:
+                bootstrap_host_tunnel.remote_cmd(['sudo', 'rm', '-rf', cluster.ssher.home_dir + '/*'])
+
+        # this method invocation looks like it is the same as the one above, and that is partially correct.
+        # the arguments to the invocation are the same, but the thing that changes is the lifecycle of the cluster
+        # the client is being created to interact with. This client is specifically for the cluster after the
+        # upgrade has taken place, and can account for any possible settings that may change for the client under
+        # the hood when it probes the cluster.
+        dcos_api_upgrade = self.dcos_api_session_factory_upgrade.apply(
+            'http://{ip}'.format(ip=cluster.masters[0].public_ip),
+            master_list,
+            master_list,
+            [h.private_ip for h in cluster.agents],
+            [h.private_ip for h in cluster.public_agents],
+            self.default_os_user)             # default_os_user
+
+        dcos_api_upgrade.wait_for_dcos()  # here we wait for DC/OS to be "up" so that we can auth this new client
+
+        self.verify_apps_state(dcos_api_upgrade, dns_app)
+
+        with logger.scope("run integration tests"):
+            # copied from test_util/test_aws_cf.py:96
```

Comment:
What's the intention of this comment? Possible refactor in future? If not, then you could omit this.